src/main/java/com/sap/oss/phosphor/fosstars/data/release/info/ReleaseInfoLoader.java

```diff
@@ -0,0 +1,110 @@
+import com.sap.oss.phosphor.fosstars.model.subject.oss.Artifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.MavenArtifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.NpmArtifact;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This data provider tries to fill out the
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature. It is based on one of the following data providers:
+ * <ul>
+ * <li>{@link ReleaseInfoFromMaven}</li>
+ * <li>{@link ReleaseInfoFromNpm}</li>
+ * </ul>
+ * The data provider uses {@link ReleasesFromGitHub} as a fallback if
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature could not be filled up from one of the above-mentioned data providers.
+ */
+public class ReleaseInfoLoader extends AbstractReleaseInfoLoader<Artifact> {
+
+  /**
+   * A {@link ReleasesFromGitHub} data provider.
+   */
+  private final DataProvider<GitHubProject> releasesFromGitHub;
+
+  /**
+   * A {@link ReleaseInfoFromMaven} data provider.
+   */
+  private final DataProvider<MavenArtifact> releaseInfoFromMaven;
+
+  /**
+   * A {@link ReleaseInfoFromNpm} data provider.
+   */
+  private final DataProvider<NpmArtifact> releaseInfoFromNpm;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param releasesFromGitHub A {@link ReleasesFromGitHub} data provider.
+   * @param releaseInfoFromMaven A {@link ReleaseInfoFromMaven} data provider.
+   * @param releaseInfoFromNpm A {@link ReleaseInfoFromNpm} data provider.
+   */
+  public ReleaseInfoLoader(ReleasesFromGitHub releasesFromGitHub,
+      ReleaseInfoFromMaven releaseInfoFromMaven, ReleaseInfoFromNpm releaseInfoFromNpm) {
+    this.releasesFromGitHub = Objects.requireNonNull(releasesFromGitHub,
+        "Oh no! You gave me a null instead of a ReleasesFromGitHub!");
+    this.releaseInfoFromMaven = Objects.requireNonNull(releaseInfoFromMaven,
+        "Oh no! You gave me a null instead of a ReleaseInfoFromMaven!");
+    this.releaseInfoFromNpm = Objects.requireNonNull(releaseInfoFromNpm,
+        "Oh no! You gave me a null instead of a ReleaseInfoFromNpm!");
+  }
+
+  @Override
+  public DataProvider<Artifact> update(Artifact artifact, ValueSet values) throws IOException {
+    Objects.requireNonNull(artifact, "Oh no! artifact cannot be null");
+
+    if (hasfeature(values)) {
+      throw new IOException(String.format(
+          "Oh no! RELEASED_ARTIFACT_VERSIONS feature is already known.",
+          artifact.getClass().getName()));
+    }
+
+    if (artifact instanceof MavenArtifact) {
+      callUpdate(releaseInfoFromMaven, (MavenArtifact) artifact, values);
+    } else if (artifact instanceof NpmArtifact) {
+      callUpdate(releaseInfoFromNpm, (NpmArtifact) artifact, values);
+    } else {
+      values.update(RELEASED_ARTIFACT_VERSIONS.unknown());
+    }
```

Comment:
If `artifact` is nether `MavenArtifact` nor `NpmArtifact` but it has `GitHubProject`, why don't we call `releaseInfoProvider`? I think `callUpdate()` is not really necessary, you can do everything in this method.

```suggestion
    if (artifact instanceof MavenArtifact) {
      releaseInfoFromMaven.update((MavenArtifact) artifact, values);
    } else if (artifact instanceof NpmArtifact) {
      releaseInfoFromNpm((NpmArtifact) artifact, values);
    }
    
    if (!hasFeatureIn(values) && artifact.project().isPresent()) {
      releasesFromGitHub.update(artifact.project().get(), values);
    }
    
    if (!hasFeatureIn(values)) {
      values.update(RELEASED_ARTIFACT_VERSIONS.unknown());
    }
```